fix: Don’t pass response to AutoRouter missing handler

istarkov opened the pull request:
### Description

The missing handler is a RequestHandler, and the response should not be passed to it.

Probaly a breaking change for non typescript users.

(_Found that prettier is not applied to the codebase, or some prettier defaults is not set see `yarn prettier`_)

### Related Issue
Link to the related issue:

### Type of Change (select one and follow subtasks)
- [ ] Documentation (README.md)
- [ ] Maintenance or repo-level work (e.g. linting, build, tests, refactoring, etc.)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
  - [ ] I have included test coverage
- [ ] Breaking change (fix or feature that would cause existing functionality/userland code to not work as expected)
  - [ ] Explain why a breaking change is necessary:
- [ ] This change requires (or is) a documentation update
  - [ ] I have added necessary local documentation (if appropriate)
  - [ ] I have added necessary [itty.dev](https://github.com/kwhitley/itty.dev) documentation (if appropriate)

istarkov committed:
Don’t pass response to missing handler

kwhitley commented:
@istarkov - this is great, and a good catch.  I'll want to add a test for this before release, so hang tight (unless you want to add that to the PR)

istarkov commented:
I’ve created a simple test to verify that the first argument of the missing function is the request.

istarkov committed:
Fix typescript config and StatusError Type imports

istarkov commented:
I also fixed the StatusError imports because they were not resolved properly when the package was imported, causing all properties of StatusError to have the any type. See the “Should give TypeScript error” comment for more details.

Example:

```ts
import { Router, type IRequest } from 'itty-router';

type CFArgs = [Env, ExecutionContext];

const router = Router<IRequest, CFArgs>({
  catch: (error) => {
    // eslint-disable-next-line no-console
    console.error(error);

    error.status satisfies number;
    error.status satisfies string; // <-- Should give typescript error
    
    return new Response(error.message, {
      status: error.status,
    });
  },
});

```

These errors will be visible if "moduleResolution": "bundler" is set in tsconfig.json under compilerOptions.

Alternatively, with the new module: "preserve" option in TypeScript 5.4, the bundler option will be implied.
Reference: [TypeScript Handbook - Module Resolution](https://www.typescriptlang.org/docs/handbook/modules/reference.html#preserve)

Additionally, the baseUrl has been removed.

So, I’ve also updated the tsconfig.json.

Splitted PR on 2 commits instead of 2 PR's.

istarkov commented:
Also found 
- https://github.com/kwhitley/itty-router/blob/7cb145879aa831f4bee1465d4b7995f659eae07f/src/types/RouterOptions.ts#L13
is a wrong type, the error is not formatted to the StatusError type

```
catch (err: any) {
        if (!other.catch) throw err
        response = await other.catch(err, request.proxy ?? request, ...args)
      }
```      

2 ways to fix, give an unknown type as in modern ts, or fomat error

kwhitley commented:
Taking a dive on this today - we've had a bit of a hurricane rip through this week :D

istarkov commented:
Interesting test had no issues with node@20.12 but fails on node@20.15. Investigating

istarkov committed:
Fix test

istarkov commented:
Fixed.

kwhitley commented:
This is great, thanks for the catch!

mootari commented:
@kwhitley Did you plan to include this fix in the changelog? It currently ends at 5.0.17: https://github.com/kwhitley/itty-router/blob/v5.0.18/CHANGELOG.md
